tests/network/localnet/test_default_bridge.py

```diff
@@ -1,35 +1,51 @@
+
 import pytest
 
 from libs.net.traffic_generator import is_tcp_connection
 from tests.network.localnet.liblocalnet import LOCALNET_BR_EX_NETWORK, client_server_active_connection
+from utilities.network import get_vmi_ip_v4_by_name
 from utilities.virt import migrate_vm_and_verify
 
 
-@pytest.mark.gating
-@pytest.mark.ipv4
-@pytest.mark.single_nic
-@pytest.mark.s390x
-@pytest.mark.usefixtures("nncp_localnet")
-@pytest.mark.polarion("CNV-11775")
-def test_connectivity_over_migration_between_localnet_vms(localnet_server, localnet_client):
-    migrate_vm_and_verify(vm=localnet_client.vm)
-    assert is_tcp_connection(server=localnet_server, client=localnet_client)
-
-
-@pytest.mark.ipv4
-@pytest.mark.single_nic
-@pytest.mark.s390x
-@pytest.mark.usefixtures("nncp_localnet")
-@pytest.mark.polarion("CNV-11925")
-def test_connectivity_post_migration_between_localnet_vms(migrated_localnet_vm, localnet_running_vms):
-    vms = list(localnet_running_vms)
-    vms.remove(migrated_localnet_vm)
-    (base_localnet_vm,) = vms
-
-    with client_server_active_connection(
-        client_vm=base_localnet_vm,
-        server_vm=migrated_localnet_vm,
-        spec_logical_network=LOCALNET_BR_EX_NETWORK,
-        port=8888,
-    ) as (client, server):
-        assert is_tcp_connection(server=server, client=client)
+class TestLocalnetVlan:
+    @pytest.mark.gating
+    @pytest.mark.ipv4
+    @pytest.mark.single_nic
+    @pytest.mark.s390x
+    @pytest.mark.usefixtures("nncp_localnet")
+    @pytest.mark.polarion("CNV-11775")
+    def test_connectivity_over_migration_between_localnet_vms(self, localnet_server, localnet_client):
+        migrate_vm_and_verify(vm=localnet_client.vm)
+        assert is_tcp_connection(server=localnet_server, client=localnet_client)
+
+    @pytest.mark.ipv4
+    @pytest.mark.single_nic
+    @pytest.mark.s390x
+    @pytest.mark.usefixtures("nncp_localnet")
+    @pytest.mark.polarion("CNV-11925")
+    def test_connectivity_post_migration_between_localnet_vms(self, migrated_localnet_vm, localnet_running_vms):
+        vms = list(localnet_running_vms)
+        vms.remove(migrated_localnet_vm)
+        (base_localnet_vm,) = vms
+
+        with client_server_active_connection(
+            client_vm=base_localnet_vm,
+            server_vm=migrated_localnet_vm,
+            spec_logical_network=LOCALNET_BR_EX_NETWORK,
+            port=8888,
+        ) as (client, server):
+            assert is_tcp_connection(server=server, client=client)
+
+
+class TestLocalnetNoVlan:
+    @pytest.mark.ipv4
+    @pytest.mark.single_nic
+    @pytest.mark.s390x
+    @pytest.mark.usefixtures("nncp_localnet")
+    @pytest.mark.polarion("CNV-12363")
+    def test_vmi_reports_ip_on_localnet_without_vlan(self, vm_localnet_no_vlan, vm_localnet_no_vlan_ip_address):
+        assert (
+            get_vmi_ip_v4_by_name(vm=vm_localnet_no_vlan, name=LOCALNET_BR_EX_NETWORK)
+            == ip_interface(vm_localnet_no_vlan_ip_address).ip
+        )
```

Comment:
same here - seeing an assert that x != y means nothing and has no context (btw, neither get_vmi_ip_v4_by_name nor ip_interface print anything)


@EdDev

Reply:
The assertion would express that an expected IP X is not what has been detected on the interface, providing the other IP that was seen or raising with an `IpNotFound` exception.

The only information which may be useful as an imidiate observation is the whole VMI interfaces status section. We can run simulations and examine this and ask the team what is missing.
@Anatw , can you post an output of different fail scenarios here? Then we will know if something is missing or not.

Reply:
The `infoSource` is the added value here as it provides the source of the IP address (is it from the guest-agent or from the pod/domain).

We should probably create some assertion helpers as a follow up.
These are probably missing all over the place.

Reply:
Ack. Marking this as resolved.